web/web.go

```diff
@@ -753,91 +686,6 @@ func (h *Handler) consoles(w http.ResponseWriter, r *http.Request) {
 	io.WriteString(w, result)
 }
 
-func (h *Handler) graph(w http.ResponseWriter, r *http.Request) {
-	h.executeTemplate(w, "graph.html", nil)
-}
-
-func (h *Handler) status(w http.ResponseWriter, r *http.Request) {
-	status := struct {
-		Birth               time.Time
-		CWD                 string
-		Version             *PrometheusVersion
-		Alertmanagers       []*url.URL
-		GoroutineCount      int
-		GOMAXPROCS          int
-		GOGC                string
-		GODEBUG             string
-		CorruptionCount     int64
-		ChunkCount          int64
-		TimeSeriesCount     int64
-		LastConfigTime      time.Time
-		ReloadConfigSuccess bool
-		StorageRetention    string
-		NumSeries           uint64
-		MaxTime             int64
-		MinTime             int64
-		Stats               *index.PostingsStats
-		Duration            string
-	}{
-		Birth:          h.birth,
-		CWD:            h.cwd,
-		Version:        h.versionInfo,
-		Alertmanagers:  h.notifier.Alertmanagers(),
-		GoroutineCount: runtime.NumGoroutine(),
-		GOMAXPROCS:     runtime.GOMAXPROCS(0),
-		GOGC:           os.Getenv("GOGC"),
-		GODEBUG:        os.Getenv("GODEBUG"),
-	}
-
-	if h.options.TSDBRetentionDuration != 0 {
-		status.StorageRetention = h.options.TSDBRetentionDuration.String()
-	}
-	if h.options.TSDBMaxBytes != 0 {
-		if status.StorageRetention != "" {
-			status.StorageRetention = status.StorageRetention + " or "
-		}
-		status.StorageRetention = status.StorageRetention + h.options.TSDBMaxBytes.String()
-	}
-
-	metrics, err := h.gatherer.Gather()
-	if err != nil {
-		http.Error(w, fmt.Sprintf("error gathering runtime status: %s", err), http.StatusInternalServerError)
-		return
-	}
-	for _, mF := range metrics {
-		switch *mF.Name {
-		case "prometheus_tsdb_head_chunks":
-			status.ChunkCount = int64(toFloat64(mF))
-		case "prometheus_tsdb_head_series":
-			status.TimeSeriesCount = int64(toFloat64(mF))
-		case "prometheus_tsdb_wal_corruptions_total":
-			status.CorruptionCount = int64(toFloat64(mF))
-		case "prometheus_config_last_reload_successful":
-			status.ReloadConfigSuccess = toFloat64(mF) != 0
-		case "prometheus_config_last_reload_success_timestamp_seconds":
-			status.LastConfigTime = time.Unix(int64(toFloat64(mF)), 0).UTC()
-		}
-	}
-
-	startTime := time.Now().UnixNano()
-	s, err := h.localStorage.Stats("__name__")
-	if err != nil {
-		if errors.Cause(err) == tsdb.ErrNotReady {
-			http.Error(w, tsdb.ErrNotReady.Error(), http.StatusServiceUnavailable)
-			return
-		}
-		http.Error(w, fmt.Sprintf("error gathering local storage statistics: %s", err), http.StatusInternalServerError)
-		return
-	}
-	status.Duration = fmt.Sprintf("%.3f", float64(time.Now().UnixNano()-startTime)/float64(1e9))
-	status.Stats = s.IndexPostingStats
-	status.NumSeries = s.NumSeries
-	status.MaxTime = s.MaxTime
-	status.MinTime = s.MinTime
-
-	h.executeTemplate(w, "status.html", status)
-}
-
 func (h *Handler) runtimeInfo() (api_v1.RuntimeInfo, error) {
```

Comment:
Wondering whether we should also remove this method from `Handler`and just pass the necessary options into the API handler, since that's the only place where it's being used now. Doesn't have to be in this PR though, if we want to think about it more.

Reply:
yeah it makes more sense it's on the API side. But I think it would make more sense to do it in another PR.

And by looking at the code in `web.go` and in `api.go` it's quite hard to understand the difference between the two. Each are exposing http endpoint, it's quite weird from a far point of view we need both.